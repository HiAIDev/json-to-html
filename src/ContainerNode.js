import React from 'react';
import {getHtmlTag} from "./getHtmltag";
import {Node} from "./Node";

export function ContainerNode({node, config}) {
  const HtmlTag = getHtmlTag(config, node.type);
  return (
    <HtmlTag>
      {node.nodes.map((node, index) => <Node node={node} config={config} key={index}/>)}
    </HtmlTag>
  );
}
