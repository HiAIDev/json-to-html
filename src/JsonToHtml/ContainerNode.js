import React from 'react';
import {getComponent} from "./getComponent";
import {Node} from "./Node";
import {NodePropTypes} from "./NodePropTypes";

export function ContainerNode({node, config}) {
  const Component = getComponent(config, node.type);
  return (
    <Component>
      {node.nodes.map((node, index) => <Node node={node} config={config} key={index}/>)}
    </Component>
  );
}

ContainerNode.propTypes = NodePropTypes;
