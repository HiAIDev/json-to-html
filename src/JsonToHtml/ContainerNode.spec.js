import React from 'react';
import {shallow} from 'enzyme';
import {ContainerNode} from "./ContainerNode";

describe('Leaf Node', () => {

  const config = {
    paragraph: 'p',
    quote: 'blockquote',
  };

  it('should render properly', () => {
    const node = {type: "quote", nodes: []};
    const renderedNode = shallow(<ContainerNode node={node} config={config}/>);

    expect(renderedNode).toMatchSnapshot();
  });
});