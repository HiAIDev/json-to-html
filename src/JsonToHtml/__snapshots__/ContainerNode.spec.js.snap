// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaf Node should render properly 1`] = `<blockquote />`;
